packages/website/private/contents/post-011.mdx

```diff
@@ -0,0 +1,114 @@
+
+## New features / Breaking Changes
+
+### Custom Vite configuration
+
+Custom Vite configuration support has been changed. Vite configuration can now be specified through the `vite` property in `waku.config.ts`. `@vitejs/plugin-rsc` provides three environments `client`, `ssr`, and `rsc` for fine-grained control over the configuration and plugin pipeline. See [Vite Environment API documentation](https://vite.dev/guide/api-environment.html) for more details.
+
+_Before_
+
+```ts
+// vite.config.ts
+import { defineConfig } from "vite";
+
+export default defineConfig({ ... })
+```
+
+```ts
+// waku.config.ts
+import { defineConfig } from "waku/config";
+
+export default defineConfig({
+  unstable_viteConfigs: { ... },
+})
+```
+
+_After_
+
+```ts
+// waku.config.ts
+import { defineConfig } from "waku/config";
+
+export default defineConfig({
+  vite: { ... }
+})
+```
+
+_Example_
+
+```ts
+// waku.config.ts
+import { defineConfig } from 'waku/config';
+
+export default defineConfig({
+  vite: {
+    environments: {
+      // environment-specific configurations.
+      client: {
+        build: {
+          // disable client assets minification
+          minify: false,
+        },
+      },
+      ssr: {
+        optimizeeDeps: {
+          include: [
+            // handle cjs package for SSR
+          ],
+        },
+      },
+      rsc: {
+        // ...
+      },
+    },
+    plugins: [
+      // custom plugins
+      {
+        name: 'my-custom-plugin',
+        transform(code, id) {
+          // e.g. transform only on `rsc` environment
+          if (this.environment.name === 'rsc') {
+            // ...
+          }
+        },
+      },
+    ],
+  },
+});
+```
+
+### Transforming external packages
+
+Previously Waku transformed all server packages on RSC module graph while externalizing all server packages SSR module graph during development. This has been changed to transform only React-related 3rd party packages on both server module graphs to be able to discover "use server" and "use client" inside the packages. Unrelated packages are kept externalizing for better package compatibility.
+
+This change can cause a new error during SSR development if there's a CJS dependency (directly or transitively) used in client components. For example, `swr` (ESM) package uses `use-sync-external-store` (CJS) internally and the plugin will show a warning `"found non-optimized CJS dependency"`. Currently, this needs to be manually mitigated by configuring `optimizeDeps.include` such as `environments.ssr.optimizeDeps.include: ["swr"]`.
+
+## Previous Implementation
+
+The previous implementation remains available via the `--experimental-legacy-cli` flag for fallback during the transition period, for example,
+
+```bash
+waku dev --experimental-legacy-cli
+waku build --experimental-legacy-cli
+waku start --experimental-legacy-cli
+```
```

Comment:
I'm not sure if we need to show this in the announcement post. a) because it will be removed very soon, b) it's very unlikely that this is needed, and c) if we want we can write it somewhere in docs/guides.

Reply:
Removed :+1: